src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1947,6 +1947,11 @@ open_common(int (*real_open)(const char *pathname, int oflags, ...), const char
 
 	if (!is_target_path)
 		goto org_func;
+	if (oflags & O_CREAT && (oflags & O_DIRECTORY || oflags & O_PATH)) {
+		/* Create a dir is not supported. */
+		errno = ENOENT;
+		return (-1);
+	}
```

Comment:
is there any chance we can be in this function if creating a directory? i.e. should this be an assertion that this is not a dir? also what about symlinks?

Reply:
Normally mkdir()/mkdirat() should be used to create a dir. However, we have no control over users' applications.
/scratch on boro (nfs4) returns -1 (errno = ENOENT), so I did the same here.
I am not aware that there is a flag for creating a symlink in open().

Reply:
@knard-intel Thank you for your information!
This document is in the section of "BUGS". We are not necessary to follow this behavior.
Shall we actually create a directory or return -1 (errno = ENOENT) for "O_DIRECTORY"?

Reply:
> @knard-intel Thank you for your information! This document is in the section of "BUGS". We are not necessary to follow this behavior. Shall we actually create a directory or return -1 (errno = ENOENT) for "O_DIRECTORY"?

You are right, from my side returning -1 (errno = ENOENT) seems the best choice.